pkg/kdump/kdump-view.jsx

```diff
@@ -20,166 +20,203 @@
-                        <TextInput id="kdump-settings-ssh-directory" key="directory"
-                                   placeholder="/var/crash" value={directory}
-                                   data-stored={directory}
-                                   onChange={value => this.props.onChange("path", value)} />
-                    </FormGroup>
-                </>
-            );
+        if (storageLocation === "nfs") {
+            newSettings.targets.nfs.server = server;
+            newSettings.targets.nfs.export = exportPath;
         }
 
-        const targetDescription = {
-            local: _("Local filesystem"),
-            nfs: _("Remote over NFS"),
-            ssh: _("Remote over SSH"),
-        };
-        // we don't support all known storage options currently
-        const storageDest = this.state.storeDest;
-        return (
+        handleSave(newSettings)
+                .then(Dialogs.close)
+                .finally(() => setIsSaving(false))
+                .catch(error => {
+                    if (error.details) {
+                        // avoid bad summary like "systemd job RestartUnit ["kdump.service","replace"] failed with result failed"
+                        // if we have a more concrete journal
+                        setError(new Error(cockpit.format(_("Unable to save settings: $0"), String(error.details))));
+                        error.message = _("Unable to save settings");
+                        error.details = <CodeBlockCode>{ error.details }</CodeBlockCode>;
+                        setError(error);
+                    } else {
+                        // without a journal, show the error as-is
+                        setError(new Error(cockpit.format(_("Unable to save settings: $0"), String(error))));
+                    }
+                });
+    };
+
+    return (
+        <Modal position="top" variant="small" id="kdump-settings-dialog" isOpen
+               title={_("Crash dump location")}
+               onClose={Dialogs.close}
+               footer={
+                   <>
+                       <Button variant="primary"
+                               isLoading={isSaving}
+                               isDisabled={isSaving}
+                               onClick={saveSettings}>
+                           {_("Save changes")}
+                       </Button>
+                       <Button variant="link"
+                               isDisabled={isSaving}
+                               className="cancel"
+                               onClick={Dialogs.close}>
+                           {_("Cancel")}
+                       </Button>
+                   </>
+               }>
+            {error && <ModalError dialogError={typeof error == 'string' ? error : error.message}
+                                  dialogErrorDetail={typeof error == 'string' ? error : error?.details || null } />}
```

Comment:
Feels a bit convoluted. How about `error.message || error` and `error?.details` ?